### PR TITLE
Refactor battery current interface

### PR DESCRIPTION
**Pull request by WickedShell**
This is a refactor I've wanted to do for awhile, but was particularly useful for scripting which is why I did it now. This removes the extra method of `has_current(void)` from the battery monitoring library. Rather then being responsible for checking if the current is available, or getting a 0 if the index is out of range this instead uses a return value on the `current_amps()` and `consumed_mah()` calls. The main advantages of this are;
  1. No need to reliably separately poll for has_current()
  1. Scripting can utilize null punning for accessing the current, which is both easier in a script, and easier to document (if null punning isn't consistently available it is a much more frustrating experience)
  1. A number of places get to skip doing work they don't need to if the current was not available
  1. The check on each backend is much tighter then the previous versions of the checks
  1. This starts to help resolve the `nullptr` dereference problem that happens if someone only enables BATT_MONITOR 2 for example

The downside, is that all the callers must select what to do now if no data was available. At the moment I've preserved the existing behavior in most cases (pass 0 out), but a good future enhancement would be to skip the rest of the work. (IE if consumed mah data isn't available, maybe we don't need to draw the label on the OSD for example).

Other things of particular note:
  - The copter motors library changed the most, but it was repeatedly polling the library for the same static data, so I think this is quite reasonable to reduce the frequency.
  - OSD had 2 different paths for drawing consumed current on the OSD. One would do rounding, one wouldn't and the only difference was which monitor was used. I've unified the methods such that they both get turned into `Ah` after the `9999 mAh` mark.

### CODE REVIEW

**Comment by rmackay9** [comment]
The Copter and AP_Motors changes look good to me.  I think the new interface (using bool) is better and more consistent with other parts of the code as well.

File: libraries/AP_BattMonitor/AP_BattMonitor.cpp [hunk 1]
```diff
@@ -302,29 +280,32 @@ float AP_BattMonitor::voltage_resting_estimate(uint8_t instance) const
 }
 
 /// current_amps - returns the instantaneous current draw in amperes
-float AP_BattMonitor::current_amps(uint8_t instance) const {
-    if (instance < _num_instances) {
-        return state[instance].current_amps;
+bool AP_BattMonitor::current_amps(float &current, uint8_t instance) const {
+    if ((instance < _num_instances) && (drivers[instance] != nullptr) && drivers[instance]->has_current()) {
```
[the remark this answers is not in the document]
Comment (WickedShell) [reply libraries/AP_BattMonitor/AP_BattMonitor.cpp]: Yeah. That was an intentional design decision. The parameter is marked needs reboot, a lot of the rest of the battery library doesn't have the check (voltage, failsafes) etc. By not supporting this on current only we are actually more consistent with our own docs/the rest of the behavior.

**Comment by WickedShell** [comment]
I did the SITL OSD test, no issues. I need to look into the `capacity_remaining_pct` call that @OXINARF raised. I suspect (since I didn't hit it when removing has_current) that this is safe to merge now, but that we probably aren't handling the missing capacity case as well as we should. Just waiting to have the time to investigate that before merging this/doing the refactor.

**Comment by WickedShell** [comment]
The cacpity_remaining_pct is worth doing in another patch, but I don't see it being anyworse then what it already is, and this doesn't need to wait on that.

**Pull request merged**